backbone/src/main/java/org/researchstack/backbone/ui/step/layout/SurveyStepLayout.java

```diff
@@ -350,4 +361,21 @@ private void checkIfAllStepsAreOptional() {
     public StepBody getStepBody() {
         return stepBody;
     }
+
+    public void revertAllChildren(@NotNull StepResult originalResult) {
+        Map<String, Object> results = originalResult.getResults();
+        for (Map.Entry<String, Object> result : results.entrySet()) {
+            String id = result.getKey();
+            Object value = result.getValue();
+            if (value instanceof StepResult) {
+                if (stepResult.getResultForIdentifier(id) != null) {
+                    ((StepResult) stepResult.getResultForIdentifier(id)).setResult(((StepResult) value).getResult());
+                }
+            }
+        }
+    }
+
+    public Boolean isFormStep() {
```

Comment:
Does this need to be public? 

You can have a similar method for `isStepResult` (to be consistent)

Reply:
Same i will do it